lib/Net/IpAddressClassifier.php

```diff
@@ -69,9 +68,26 @@ public function isLocalAddress(string $ip): bool {
 			/* Range address */
 			return true;
 		}
-		if (IpUtils::checkIp($ip, self::LOCAL_ADDRESS_RANGES)) {
-			/* Within local range */
-			return true;
+
+		if (class_exists(\Symfony\Component\HttpFoundation\IpUtils::class)) {
+			// >= Nextcloud 25
+			if (\Symfony\Component\HttpFoundation\IpUtils::checkIp($ip, self::LOCAL_ADDRESS_RANGES)) {
+				/* Within local range */
+				return true;
+			}
+		} elseif (class_exists(\OC\Http\IpUtils::class)) {
+			// >= Nextcloud 24.0.4
+			// >= Nextcloud 23.0.8
+			// >= Nextcloud 22.2.11
+			if (\OC\Http\IpUtils::checkIp($ip, self::LOCAL_ADDRESS_RANGES)) {
+				/* Within local range */
+				return true;
+			}
+		} else {
+			// < Nextcloud 24.0.4
+			// < Nextcloud 23.0.8
+			// < Nextcloud 22.2.11
+			// IpUtils is not available.
 		}
```

Comment:
can't you remove that else block since it will return false below anyway?

Reply:
done done